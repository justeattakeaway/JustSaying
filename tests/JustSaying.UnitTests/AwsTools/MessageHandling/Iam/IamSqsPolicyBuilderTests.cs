using JustSaying.AwsTools.MessageHandling;
using Newtonsoft.Json.Linq;

namespace JustSaying.UnitTests.AwsTools.MessageHandling.Iam;

public class IamSqsPolicyBuilderTests
{
    [Fact]
    public void ShouldGenerateApprovedIamPolicy()
    {
        // arrange
        var sqsPolicyDetails = new SqsPolicyDetails
        {
            SourceArn = "arn:aws:sqs:ap-southeast-2:123456789012:topic",
        };

        // act
        var policy = IamSqsPolicyBuilder.BuildPolicyJson(sqsPolicyDetails);

        // assert
        policy.ShouldMatchApproved(c =>
        {
            c.SubFolder("Approvals");
            // Sids are generated from guids on each invocation so must be ignored
            // when performing approval tests
            c.WithScrubber(ScrubSids);
        });
    }

    [Fact]
    public void ShouldGenerateApprovedIamPolicyWithWildcardFromEmptySourceArn()
    {
        // arrange
        var sqsPolicyDetails = new SqsPolicyDetails
        {
            SourceArn = "",
        };

        // act
        var policy = IamSqsPolicyBuilder.BuildPolicyJson(sqsPolicyDetails);

        // assert
        policy.ShouldMatchApproved(c =>
        {
            c.SubFolder("Approvals");
            c.WithScrubber(ScrubSids);
        });
    }

    private static string ScrubSids(string iamPolicy)
    {
        // Sids are generated from guids on each invocation so must be ignored
        // when performing approval tests
        var json = JObject.Parse(iamPolicy);
        return iamPolicy
            .Replace(json["Statement"]![0]!["Sid"]!.ToString(), "<sid>");
    }
}
