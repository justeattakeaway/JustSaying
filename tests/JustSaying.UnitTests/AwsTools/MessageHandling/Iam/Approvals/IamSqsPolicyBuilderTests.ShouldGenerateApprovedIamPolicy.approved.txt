{
    "Version" : "2012-10-17",
    "Statement" : [
        {
            "Sid" : "<sid>",
            "Effect" : "Allow",
            "Principal" : {
                "AWS" : "*"
            },
            "Action"    : "sqs:SendMessage",
            "Resource"  : "",
            "Condition" : {
                "ArnLike" : {
                    "aws:SourceArn" : "arn:aws:sqs:ap-southeast-2:123456789012:*"
                }
            }
        }
    ]
}